### PR TITLE
Mock topology testing

### PR DESCRIPTION
**Pull request by pohly**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:

For testing the upcoming "unset selected node" feature in external-provisioner.

**Which issue(s) this PR fixes**:

Related-to: https://github.com/kubernetes/kubernetes/issues/72031

**Special notes for your reviewer**:

This could be merged as-is, but it might also make sense to first review the other PRs.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: mock/service/controller.go [hunk 1]
```diff
@@ -58,6 +63,18 @@ func (s *service) CreateVolume(
 		return nil, status.Errorf(codes.OutOfRange, "Requested capacity %d exceeds maximum allowed %d", capacity, MaxStorageCapacity)
 	}
 
+	// Simulate "out of storage space" error?
+	if capacityLimit := req.GetParameters()["capacityLimit"]; capacityLimit != "" {
```
Comment (msau42) [inline mock/service/controller.go]: @jsafrane how soon is that error injection coming?

Comment (pohly) [reply mock/service/controller.go]: /hold

There's now a WIP PR for that in PR #251. Using that approach makes more sense than adding special cases like this one here to the code.

**Comment by pohly** [comment]
I've stripped out the error injection and rewrote https://github.com/kubernetes/kubernetes/pull/88114 to use JavaScript error injection.

What's left is just the optional topology support. Can we merge this now? Then I'll do a csi-test release.

**Comment by pohly** [comment]
That should also unblock https://github.com/kubernetes/kubernetes/pull/89041.

File: mock/service/controller.go (hunk 1 above)
Comment (pohly) [reply mock/service/controller.go]: /hold cancel

**Pull request merged**